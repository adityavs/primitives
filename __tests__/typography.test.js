import React from 'react';
import renderer from 'react-test-renderer';

import { BaseText, H1, ConfigProvider } from '../index'

describe('<BaseText>', () => {
  it('should render Alan Turing; default formatting', () => {
    const tree = renderer
      .create(
          <ConfigProvider>
            <BaseText>Alan Turing</BaseText>
          </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Alan Turing; bold', () => {
    const tree = renderer
      .create(
          <ConfigProvider>
            <BaseText bold>Alan Turing</BaseText>
          </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Alan Turing; uppercase underline strikeout semibold center', () => {
    const tree = renderer
      .create(
          <ConfigProvider>
            <BaseText uppercase underline strikeout semibold center>Alan Turing</BaseText>
          </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Alan Turing; fontWeight={thin} fontSize={18} color={#FFF}', () => {
    const tree = renderer
      .create(
          <ConfigProvider>
            <BaseText fontWeight='thin' fontSize={18} color='#FFF'>Alan Turing</BaseText>
          </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe('<H1>', () => {
  it('should render Alan Turing', () => {
    const tree = renderer
      .create(
          <ConfigProvider>
            <H1>Alan Turing</H1>
          </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
