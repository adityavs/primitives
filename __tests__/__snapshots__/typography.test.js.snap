// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseText> should render Alan Turing; bold 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "Helvetica",
        "fontWeight": "400",
      },
      Object {
        "fontWeight": "700",
      },
      Object {},
    ]
  }
>
  Alan Turing
</Text>
`;

exports[`<BaseText> should render Alan Turing; default formatting 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "Helvetica",
        "fontWeight": "400",
      },
      Object {},
    ]
  }
>
  Alan Turing
</Text>
`;

exports[`<BaseText> should render Alan Turing; fontWeight={thin} fontSize={18} color={#FFF} 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "Helvetica",
        "fontWeight": "400",
      },
      Object {
        "fontWeight": "200",
      },
      Object {
        "fontSize": 18,
      },
      Object {
        "color": "#FFF",
      },
      Object {},
    ]
  }
>
  Alan Turing
</Text>
`;

exports[`<BaseText> should render Alan Turing; uppercase underline strikeout semibold center 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "Helvetica",
        "fontWeight": "400",
      },
      Object {
        "textTransform": "uppercase",
      },
      Object {
        "textDecorationLine": "underline",
      },
      Object {
        "textDecorationLine": "line-through",
      },
      Object {
        "fontWeight": "600",
      },
      Object {
        "textAlign": "center",
      },
      Object {},
    ]
  }
>
  Alan Turing
</Text>
`;

exports[`<H1> should render Alan Turing 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "Helvetica",
        "fontWeight": "400",
      },
      Object {
        "fontSize": 32,
      },
      Object {},
    ]
  }
>
  Alan Turing
</Text>
`;
